### PR TITLE
prep for task planning predicate invention: pass train_tasks through to heuristics, and expose task planning only

### PR DESCRIPTION
**Pull request by ronuchit**

### CODE REVIEW

File: tests/test_planning.py [hunk 1]
```diff
@@ -21,8 +22,16 @@ def test_sesame_plan():
     nsrts = get_gt_nsrts(env.predicates, env.options)
     task = next(env.train_tasks_generator())[0]
     option_model = create_option_model(CFG.option_model_name, env.simulate)
-    plan = sesame_plan(task, option_model, nsrts, env.predicates, 1, 123)
+    plan, _ = sesame_plan(task, option_model, nsrts, env.predicates,
```
Comment (tomsilver) [inline tests/test_planning.py:25]: I'm confused, was this a bug in the old unit test?

**Review by tomsilver — APPROVED** [review]
thanks for doing this

File: tests/test_planning.py (hunk 1 above)
Comment (ronuchit) [reply tests/test_planning.py:25]: yes lol the metrics return wasn't being considered

File: src/planning.py [hunk 2]
```diff
@@ -126,6 +135,8 @@ def _run_search(task: Task,
         if task.goal.issubset(node.atoms):
             # If this skeleton satisfies the goal, run low-level search.
             metrics["num_skeletons_optimized"] += 1
+            if not do_low_level_search:
```
[the remark this answers is not in the document]
Comment (ronuchit) [reply src/planning.py:138]: that would be very tough! let's leave it...

**Pull request merged**